diff --git a/chrome/browser/ui/views/location_bar/location_bar_view.cc b/chrome/browser/ui/views/location_bar/location_bar_view.cc
index 08cf89b6fa779c484b686940c30c3969ed30c869..b01f17e9a0d541aca3a9c594c2c1f5ce7b3f34df 100644
--- a/chrome/browser/ui/views/location_bar/location_bar_view.cc
+++ b/chrome/browser/ui/views/location_bar/location_bar_view.cc
@@ -412,7 +412,7 @@ gfx::Size LocationBarView::CalculatePreferredSize() const {
   return min_size;
 }
 
-void LocationBarView::Layout() {
+void LocationBarView::Layout(views::View* right_most) {
   if (!IsInitialized())
     return;
 
@@ -506,6 +506,11 @@ void LocationBarView::Layout() {
     }
   };
 
+  if (right_most && right_most->visible())
+    trailing_decorations.AddDecoration(0,
+                              height(),
+                              false, 0, edge_padding, right_most);
+
   if (star_view_)
     add_trailing_decoration(star_view_);
   add_trailing_decoration(page_action_icon_container_view_);
