diff --git a/chrome/common/pref_names.h b/chrome/common/pref_names.h
index 37b9403281d97b2123dfd4b754574da8f0441fb2..508e9259f4cae2ab9131116ac68684b29db4245c 100644
--- a/chrome/common/pref_names.h
+++ b/chrome/common/pref_names.h
@@ -359,6 +359,8 @@ extern const char kImportDialogBookmarks[];
 extern const char kImportDialogHistory[];
 extern const char kImportDialogSavedPasswords[];
 extern const char kImportDialogSearchEngine[];
+extern const char kImportDialogCookies[];
+extern const char kImportDialogStats[];
 
 extern const char kProfileAvatarIndex[];
 extern const char kProfileUsingDefaultName[];
