diff --git a/chrome/browser/ui/webui/settings/md_settings_localized_strings_provider.cc b/chrome/browser/ui/webui/settings/md_settings_localized_strings_provider.cc
index 55bfa5a287c257bc9c8d0b1616a6c1223e4d24de..98412f03e99dadd596ba0b481ad07dad859f3f70 100644
--- a/chrome/browser/ui/webui/settings/md_settings_localized_strings_provider.cc
+++ b/chrome/browser/ui/webui/settings/md_settings_localized_strings_provider.cc
@@ -2794,6 +2794,7 @@ void AddLocalizedStrings(content::WebUIDataSource* html_source,
 #endif
   policy_indicator::AddLocalizedStrings(html_source);
 
+  BraveAddLocalizedStrings(html_source, profile);
   html_source->SetJsonPath(kLocalizedStringsFile);
 }
 
