diff --git a/components/sync/engine_impl/sync_manager_impl.h b/components/sync/engine_impl/sync_manager_impl.h
index 4a6f7f40110d0f38bf40a994c0c5eb90535d9af5..d7dc64093adef66f9cf89cb98cc2212b2a2c64fc 100644
--- a/components/sync/engine_impl/sync_manager_impl.h
+++ b/components/sync/engine_impl/sync_manager_impl.h
@@ -138,6 +138,9 @@ class SyncManagerImpl
   void OnBackedOffTypesChanged(ModelTypeSet backed_off_types) override;
   void OnMigrationRequested(ModelTypeSet types) override;
   void OnProtocolEvent(const ProtocolEvent& event) override;
+  void OnNudgeSyncCycle() override;
+  void OnPollSyncCycle(brave_sync::GetRecordsCallback cb,
+                       base::WaitableEvent* wevent) override;
 
   // ServerConnectionEventListener implementation.
   void OnServerConnectionEvent(const ServerConnectionEvent& event) override;
