diff --git a/components/sync/engine_impl/sync_manager_impl.h b/components/sync/engine_impl/sync_manager_impl.h
index 4a6f7f40110d0f38bf40a994c0c5eb90535d9af5..960f62416498ee98a0c230174abea8d3a8e334ba 100644
--- a/components/sync/engine_impl/sync_manager_impl.h
+++ b/components/sync/engine_impl/sync_manager_impl.h
@@ -138,6 +138,8 @@ class SyncManagerImpl
   void OnBackedOffTypesChanged(ModelTypeSet backed_off_types) override;
   void OnMigrationRequested(ModelTypeSet types) override;
   void OnProtocolEvent(const ProtocolEvent& event) override;
+  void OnNudgeSyncCycle() override;
+  void OnPollSyncCycle() override;
 
   // ServerConnectionEventListener implementation.
   void OnServerConnectionEvent(const ServerConnectionEvent& event) override;
