diff --git a/components/sync/driver/glue/sync_backend_host_core.h b/components/sync/driver/glue/sync_backend_host_core.h
index dced9de5a2081185acb22c900e69e90080e54cdf..08c68a1f6d49bf4876fae701ae4365652c0ffb56 100644
--- a/components/sync/driver/glue/sync_backend_host_core.h
+++ b/components/sync/driver/glue/sync_backend_host_core.h
@@ -59,6 +59,8 @@ class SyncBackendHostCore
   void OnActionableError(const SyncProtocolError& sync_error) override;
   void OnMigrationRequested(ModelTypeSet types) override;
   void OnProtocolEvent(const ProtocolEvent& event) override;
+  void OnNudgeSyncCycle() override;
+  void OnPollSyncCycle(brave_sync::GetRecordsCallback cb) override;
 
   // TypeDebugInfoObserver implementation
   void OnCommitCountersUpdated(ModelType type,
