diff --git a/third_party/blink/renderer/core/dom/text.cc b/third_party/blink/renderer/core/dom/text.cc
index 7ee5d04f48a83d8454dcdbbf124ee296e7f10d48..d257347d7dabc17d12c69833d0b045c473b24c42 100644
--- a/third_party/blink/renderer/core/dom/text.cc
+++ b/third_party/blink/renderer/core/dom/text.cc
@@ -43,10 +43,26 @@
 #include "third_party/blink/renderer/platform/wtf/text/cstring.h"
 #include "third_party/blink/renderer/platform/wtf/text/string_builder.h"
 
+#include "brave/third_party/blink/brave_page_graph/buildflags/buildflags.h"
+#if BUILDFLAG(BRAVE_PAGE_GRAPH_ENABLED)
+#include "third_party/blink/renderer/core/dom/document.h"
+#include "third_party/blink/renderer/core/dom/dom_node_ids.h"
+#include "brave/third_party/blink/brave_page_graph/page_graph.h"
+#endif
+
 namespace blink {
 
 Text* Text::Create(Document& document, const String& data) {
+#if BUILDFLAG(BRAVE_PAGE_GRAPH_ENABLED)
+  Text* text_node = MakeGarbageCollected<Text>(document, data, kCreateText);
+  if (document.GetPageGraph() != nullptr) {
+    document.GetPageGraph()->RegisterHTMLTextNodeCreated(
+      DOMNodeIds::IdForNode(text_node), data);
+  }
+  return text_node;
+#else
   return MakeGarbageCollected<Text>(document, data, kCreateText);
+#endif
 }
 
 Text* Text::CreateEditingText(Document& document, const String& data) {
