diff --git a/third_party/blink/renderer/core/html/html_frame_owner_element.cc b/third_party/blink/renderer/core/html/html_frame_owner_element.cc
index 9eb9c7c7c9d37973879d18b3bcfa275edf5a0ae2..8d6653c08f45fd346f7231f9c77f2862f49ff7a8 100644
--- a/third_party/blink/renderer/core/html/html_frame_owner_element.cc
+++ b/third_party/blink/renderer/core/html/html_frame_owner_element.cc
@@ -50,6 +50,12 @@
 #include "third_party/blink/renderer/platform/runtime_enabled_features.h"
 #include "third_party/blink/renderer/platform/weborigin/security_origin.h"
 
+#include "brave/third_party/blink/brave_page_graph/buildflags/buildflags.h"
+#if BUILDFLAG(BRAVE_PAGE_GRAPH_ENABLED)
+#include "third_party/blink/renderer/core/dom/document.h"
+#include "third_party/blink/renderer/core/dom/dom_node_ids.h"
+#endif
+
 namespace blink {
 
 namespace {
@@ -384,6 +390,13 @@ bool HTMLFrameOwnerElement::LoadOrRedirectSubframe(
 
   LocalFrame* child_frame =
       GetDocument().GetFrame()->Client()->CreateFrame(frame_name, this);
+#if BUILDFLAG(BRAVE_PAGE_GRAPH_ENABLED)
+  if (GetDocument().GetPageGraph() != nullptr) {
+    const DOMNodeId dom_node_id = DOMNodeIds::IdForNode(this);
+    child_frame->SetPageGraphNodeIdInParentDocument(dom_node_id);
+  }
+#endif
+
   DCHECK_EQ(ContentFrame(), child_frame);
   if (!child_frame)
     return false;
