diff --git a/third_party/blink/renderer/core/dom/document.cc b/third_party/blink/renderer/core/dom/document.cc
index e5d02bd5288267372a37bb4924d144abcf9a312c..d8ef07be91dbdb79ae09ba03aee091bde4360709 100644
--- a/third_party/blink/renderer/core/dom/document.cc
+++ b/third_party/blink/renderer/core/dom/document.cc
@@ -301,6 +301,11 @@ using WeakDocumentSet = blink::HeapHashSet<blink::WeakMember<blink::Document>>;
 static WeakDocumentSet& liveDocumentSet();
 #endif
 
+#include "brave/third_party/blink/brave_page_graph/buildflags/buildflags.h"
+#if BUILDFLAG(BRAVE_PAGE_GRAPH_ENABLED)
+#include "brave/third_party/blink/brave_page_graph/page_graph.h"
+#endif
+
 namespace blink {
 
 using namespace html_names;
@@ -730,7 +735,6 @@ Document::Document(const DocumentInit& initializer,
         &FetchContext::NullInstance(), GetTaskRunner(TaskType::kNetworking)));
   }
   DCHECK(fetcher_);
-
   root_scroller_controller_ = RootScrollerController::Create(*this);
 
   // We depend on the url getting immediately set in subframes, but we
@@ -770,6 +774,10 @@ Document::Document(const DocumentInit& initializer,
 #ifndef NDEBUG
   liveDocumentSet().insert(this);
 #endif
+
+#if BUILDFLAG(BRAVE_PAGE_GRAPH_ENABLED)
+  page_graph_ = nullptr;
+#endif
 }
 
 Document::~Document() {
@@ -780,7 +788,23 @@ Document::~Document() {
   DCHECK(!ax_object_cache_);
 
   InstanceCounters::DecrementCounter(InstanceCounters::kDocumentCounter);
+#if BUILDFLAG(BRAVE_PAGE_GRAPH_ENABLED)
+  if (page_graph_ != nullptr) {
+    delete page_graph_;
+    page_graph_ = nullptr;
+  }
+#endif
+}
+
+#if BUILDFLAG(BRAVE_PAGE_GRAPH_ENABLED)
+brave_page_graph::PageGraph* Document::GetPageGraph() {
+  if (page_graph_ == nullptr && Url().ProtocolIsInHTTPFamily()) {
+    page_graph_ = new brave_page_graph::PageGraph();
+  }
+  return page_graph_;
 }
+#endif
+
 
 Range* Document::CreateRangeAdjustedToTreeScope(const TreeScope& tree_scope,
                                                 const Position& position) {
