diff --git a/third_party/blink/renderer/modules/credentialmanager/BUILD.gn b/third_party/blink/renderer/modules/credentialmanager/BUILD.gn
index 18f143a1e7c1a8bfb9ecf8d2b2702195c0065488..d530694993fa2082f8d463f111f57543da237613 100644
--- a/third_party/blink/renderer/modules/credentialmanager/BUILD.gn
+++ b/third_party/blink/renderer/modules/credentialmanager/BUILD.gn
@@ -19,6 +19,7 @@ blink_modules_sources("credentialmanager") {
     "credential_manager_type_converters.cc",
     "credential_manager_type_converters.h",
     "credentials_container.cc",
+    "//brave/chromium_src/third_party/blink/render/modules/credentialmanager/brave_credentials_container.cc",
     "credentials_container.h",
     "federated_credential.cc",
     "federated_credential.h",
