diff --git a/third_party/blink/renderer/core/html/lazy_load_frame_observer.cc b/third_party/blink/renderer/core/html/lazy_load_frame_observer.cc
index 40d89cf594a5d6068fc9d609f29d9a8e6c80d51a..94186b9d2a3247bc4aba2c819f4722174859b62d 100644
--- a/third_party/blink/renderer/core/html/lazy_load_frame_observer.cc
+++ b/third_party/blink/renderer/core/html/lazy_load_frame_observer.cc
@@ -28,6 +28,13 @@
 #include "third_party/blink/renderer/platform/wtf/assertions.h"
 #include "third_party/blink/renderer/platform/wtf/vector.h"
 
+#include "brave/third_party/blink/brave_page_graph/buildflags/buildflags.h"
+#if BUILDFLAG(BRAVE_PAGE_GRAPH_ENABLED)
+#include "third_party/blink/renderer/core/dom/document.h"
+#include "third_party/blink/renderer/core/dom/dom_node_ids.h"
+#include "brave/third_party/blink/brave_page_graph/page_graph.h"
+#endif
+
 namespace blink {
 
 namespace {
@@ -178,6 +185,15 @@ void LazyLoadFrameObserver::LoadImmediately() {
   // DisconnectContentFrame() if the content frame changes.
   DCHECK(element_->ContentFrame());
 
+#if BUILDFLAG(BRAVE_PAGE_GRAPH_ENABLED)
+  if (element_->GetDocument().GetPageGraph() != nullptr) {
+    element_->GetDocument().GetPageGraph()->RegisterRequestStartFromElm(
+      DOMNodeIds::IdForNode(element_),
+      scoped_request_info->resource_request.Url().GetString().Utf8().data(),
+      brave_page_graph::kRequestTypeDocument);
+  }
+#endif
+
   // Note that calling FrameLoader::StartNavigation() causes the
   // |lazy_load_intersection_observer_| to be disconnected.
   ToLocalFrame(element_->ContentFrame())
