diff --git a/third_party/blink/renderer/core/dom/document.cc b/third_party/blink/renderer/core/dom/document.cc
index e5d02bd5288267372a37bb4924d144abcf9a312c..60805970292ba9550c566a5233ea823e27de5e75 100644
--- a/third_party/blink/renderer/core/dom/document.cc
+++ b/third_party/blink/renderer/core/dom/document.cc
@@ -27,6 +27,8 @@
  * Boston, MA 02110-1301, USA.
  */
 
+#include <iostream>
+
 #include "third_party/blink/renderer/core/dom/document.h"
 
 #include <memory>
@@ -301,6 +303,11 @@ using WeakDocumentSet = blink::HeapHashSet<blink::WeakMember<blink::Document>>;
 static WeakDocumentSet& liveDocumentSet();
 #endif
 
+#include "brave/third_party/blink/brave_page_graph/buildflags/buildflags.h"
+#if BUILDFLAG(BRAVE_PAGE_GRAPH_ENABLED)
+#include "brave/third_party/blink/brave_page_graph/page_graph.h"
+#endif
+
 namespace blink {
 
 using namespace html_names;
@@ -730,7 +737,6 @@ Document::Document(const DocumentInit& initializer,
         &FetchContext::NullInstance(), GetTaskRunner(TaskType::kNetworking)));
   }
   DCHECK(fetcher_);
-
   root_scroller_controller_ = RootScrollerController::Create(*this);
 
   // We depend on the url getting immediately set in subframes, but we
@@ -767,9 +773,18 @@ Document::Document(const DocumentInit& initializer,
   // loading resources ignoring the defersLoading flag.
   DCHECK(!ParentDocument() || !ParentDocument()->IsContextPaused());
 
+  if (Url().ProtocolIsInHTTPFamily()) {
+    std::cout << "\n\n\n\n\n\n";
+    std::cout << "Init document: " << Url().GetString().Utf8().data() << "\n";
+  }
+
 #ifndef NDEBUG
   liveDocumentSet().insert(this);
 #endif
+
+#if BUILDFLAG(BRAVE_PAGE_GRAPH_ENABLED)
+  page_graph_ = nullptr;
+#endif
 }
 
 Document::~Document() {
@@ -780,7 +795,31 @@ Document::~Document() {
   DCHECK(!ax_object_cache_);
 
   InstanceCounters::DecrementCounter(InstanceCounters::kDocumentCounter);
+#if BUILDFLAG(BRAVE_PAGE_GRAPH_ENABLED)
+  if (page_graph_ != nullptr) {
+    delete page_graph_;
+    page_graph_ = nullptr;
+  }
+#endif
+}
+
+#if BUILDFLAG(BRAVE_PAGE_GRAPH_ENABLED)
+brave_page_graph::PageGraph* Document::GetPageGraph() {
+  if (page_graph_ == nullptr && Url().ProtocolIsInHTTPFamily()) {
+    std::cout << "GetPageGraph: " << Url().GetString().Utf8().data() << std::endl;
+    page_graph_ = new brave_page_graph::PageGraph();
+    HTMLCollection* all_items = all();
+    unsigned num_items = all_items->length();
+    for (unsigned i = 0; i < num_items; i += 1) {
+      Element* elm = all_items->item(i);
+      page_graph_->RegisterHTMLElementNodeCreated(
+        DOMNodeIds::IdForNode(elm), elm->tagName().Utf8().data());
+    }
+  }
+  return page_graph_;
 }
+#endif
+
 
 Range* Document::CreateRangeAdjustedToTreeScope(const TreeScope& tree_scope,
                                                 const Position& position) {
